pkg/splunk/client/awss3client.go

```diff
@@ -34,9 +40,48 @@ func getRegion(endpoint string) string {
 	return pattern.FindStringSubmatch(endpoint)[1]
 }
 
+// InitAWSClientWrapper is a wrapper around InitClientSession
+func InitAWSClientWrapper(region, accessKeyID, secretAccessKey string, isNotInTestContext *bool) interface{} {
+	return InitAWSClientSession(region, accessKeyID, secretAccessKey, isNotInTestContext)
+}
+
+// InitAWSClientSession initializes and returns a client session object
+func InitAWSClientSession(region, accessKeyID, secretAccessKey string, isNotInTestContext *bool) SplunkAWSS3Client {
+	scopedLog := log.WithName("InitAWSClientSession")
+
+	*isNotInTestContext = true
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region),
+		Credentials: credentials.NewStaticCredentials(
+			accessKeyID,     // id
+			secretAccessKey, // secret
+			"")},            // token
+	)
+	if err != nil {
+		scopedLog.Error(err, "Failed to initialize an AWS S3 session.")
+		return nil
+	}
+
+	return s3.New(sess)
+}
+
 // NewAWSS3Client returns an AWS S3 client
-func NewAWSS3Client(bucketName string, accessKeyID string, secretAccessKey string, prefix string, startAfter string, endpoint string) S3Client {
+func NewAWSS3Client(bucketName string, accessKeyID string, secretAccessKey string, prefix string, startAfter string, endpoint string, fn func(string, string, string, *bool) interface{}) (S3Client, error) {
+	var isNotInTestContext bool
```

Comment:
Can we change this name to `isMockClient` rather that `isNot...`.  Makes it confusing.